### PR TITLE
Create and handle PermissionsRequest

### PR DESCRIPTION
**Pull request by diehuxx**
First PR implementing [Technical Proposal 26](https://github.com/TBD54566975/technical-proposals/pull/10)

## Changes
* Update JSON schema for `PermissionsRequest` and `PermissionsGrant`
* Add handler and tests for `PermissionsRequest`
* Cut out outdated message structure from `PermissionsGrant`. This PR doesn't test creation or handling of `PermissionsGrant` messages. That'll come in the next one.
* Reduce `scope` and `conditions` in JSON schema to bare bones. I'll add scope and conditions options back in as I implement the functionality of each one.

### CODE REVIEW

File: src/interfaces/permissions-request.ts [hunk 1]
```diff
@@ -3,18 +3,18 @@ import type { PermissionConditions, PermissionScope } from '../types/permissions
...
 import { getCurrentTimeInHighPrecision } from '../utils/time.js';
-import { v4 as uuidv4 } from 'uuid';
+import { removeUndefinedProperties } from '../utils/object.js';
 import { validateAuthorizationIntegrity } from '../core/auth.js';
 import { DwnInterfaceName, DwnMethodName, Message } from '../core/message.js';
 
 type PermissionsRequestOptions = {
   dateCreated?: string;
-  conditions?: PermissionConditions;
   description: string;
```
Comment (thehenrytsai) [inline src/interfaces/permissions-request.ts:12]: Sanity: a bit odd that `description` in `descriptor` is optional but we require it in the SDK. Should we make it optional for consistency?

File: tests/utils/test-data-generator.ts [hunk 2]
```diff
@@ -473,18 +494,27 @@ export class TestDataGenerator {
...
-      description                 : 'drugs',
-      grantedBy                   : 'did:jank:bob',
-      grantedTo                   : 'did:jank:alice',
-      scope                       : { method: 'RecordsWrite' },
-      authorizationSignatureInput : { privateJwk: privateJwk, protectedHeader: { alg: privateJwk.alg as string, kid: 'whatev' } }
+  public static async generatePermissionsRequest(input?: GeneratePermissionsRequestInput): Promise<GeneratePermissionsRequestOutput> {
+    const author = input?.author ?? await TestDataGenerator.generatePersona();
+    const permissionsRequest = await PermissionsRequest.create({
+      dateCreated : getCurrentTimeInHighPrecision(),
+      description : input?.description ?? 'drugs',
```
Comment (thehenrytsai) [inline tests/utils/test-data-generator.ts:501]: If you took the other comment, then we can get rid of the drug reference here. 🤣 
```suggestion
      description : input?.description,
```

File: src/interfaces/permissions-request.ts (hunk 1 above)
Comment (diehuxx) [reply src/interfaces/permissions-request.ts]: Ope, good catch. Updated to make it optional everywhere.

**Review by thehenrytsai — APPROVED** [review]
🎊 🙏 🚀

**Pull request merged**